mongo/integration/collection_test.go

```diff
@@ -1154,6 +1154,59 @@ func TestCollection(t *testing.T) {
 				mt.Fatalf("expected BulkWrite error %v, got %v", mongo.ErrUnacknowledgedWrite, err)
 			}
 		})
+		mt.Run("insert and delete with batches", func(mt *mtest.T) {
+			// // TODO(GODRIVER-425): remove this as part a larger project to
+			// // refactor integration and other longrunning tasks.
+			if os.Getenv("EVR_TASK_ID") == "" {
+				mt.Skip("skipping long running integration test outside of evergreen")
+			}
+
+			// grouped together because delete requires the documents to be inserted
+			numDocs := 100050
+			var insertModels []mongo.WriteModel
+			var deleteModels []mongo.WriteModel
+			for i := 0; i < numDocs; i++ {
+				d := bson.D{
+					{"a", int32(i)},
+					{"b", int32(i * 2)},
+					{"c", int32(i * 3)},
+				}
+				insertModels = append(insertModels, mongo.NewInsertOneModel().SetDocument(d))
+				deleteModels = append(deleteModels, mongo.NewDeleteOneModel().SetFilter(bson.D{}))
+			}
+			res, err := mt.Coll.BulkWrite(context.Background(), insertModels)
+			assert.Nil(mt, err, "BulkWrite error: %v", err)
+			assert.Equal(mt, int64(numDocs), res.InsertedCount, "expected %v inserted documents, got %v", numDocs, res.InsertedCount)
```

Comment:
We also need to assert that the `BulkWrite` call sends multiple requests. Otherwise, we don't know if the results are correct because the numbers were merged correctly or because we incorrectly sent them as a single request.

Reply:
done